From acb81624b4f50fed52cb1b3829809ee2f7377093 Mon Sep 17 00:00:00 2001
From: Venkatarama Avadhani <venkatarama.avadhani@ittiam.com>
Date: Fri, 11 Aug 2023 15:19:24 +0000
Subject: [PATCH] Initialise VPS buffer to NULL in constructor

Missing initialisation of this pointer could lead to an incorrect free
if the ARTWriter object is cleared immeddiately after the constructor
call.

Bug: 287298721
Test: rtp_writer_fuzzer
(cherry picked from https://partner-android-review.googlesource.com/q/commit:2710696b001f2e95586151c1ee337a4e3c4da48a)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:900195c1d3589c7cbf9e116f61bebaefc0519101)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:0efe2b4d6b739650039c2cab176ef11d5f5ac49c)
Merged-In: I08eacd7a0201bc9a41b821e20cae916d8870147a
Change-Id: I08eacd7a0201bc9a41b821e20cae916d8870147a
---
 media/libstagefright/rtsp/ARTPWriter.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/libstagefright/rtsp/ARTPWriter.cpp b/media/libstagefright/rtsp/ARTPWriter.cpp
index 8990f0cf7c..e20d70235b 100644
--- a/media/libstagefright/rtsp/ARTPWriter.cpp
+++ b/media/libstagefright/rtsp/ARTPWriter.cpp
@@ -105,6 +105,7 @@ ARTPWriter::ARTPWriter(int fd)
 
     mRTCPAddr = mRTPAddr;
     mRTCPAddr.sin_port = htons(ntohs(mRTPAddr.sin_port) | 1);
+    mVPSBuf = NULL;
     mSPSBuf = NULL;
     mPPSBuf = NULL;
 
-- 
2.42.0.515.g380fc7ccd1-goog

